networking_cisco/tests/unit/cisco/cfg_agent/test_aci_asr1k_cfg_syncer.py

```diff
@@ -0,0 +1,112 @@
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from oslo_config import cfg
+from oslo_serialization import jsonutils
+
+import mock
+from networking_cisco.tests import base
+from networking_cisco.tests.unit.cisco.cfg_agent import (
+    test_asr1k_cfg_syncer as test_sync)
+
+
+from networking_cisco.plugins.cisco.cfg_agent.device_drivers.asr1k import (
+    aci_asr1k_cfg_syncer)
+
+
+class AciASR1kCfgSyncer(test_sync.ASR1kCfgSyncer):
+
+    def _read_neutron_db_data(self):
+        """
+        helper function for reading the dummy neutron router db
+        """
+        with open(base.ROOTDIR +
+                  '/unit/cisco/etc/cfg_syncer/neutron_router_db_aci.json',
+                  'r') as fp:
+            self.router_db_info = jsonutils.load(fp)
+
+    def setUp(self):
+        super(AciASR1kCfgSyncer, self).setUp()
+
+        self.config_syncer = aci_asr1k_cfg_syncer.ConfigSyncer(
+            self.router_db_info, self.driver, self.hosting_device_info)
+
+    def tearDown(self):
+        super(AciASR1kCfgSyncer, self).tearDown()
+
+    def test_delete_invalid_cfg_empty_routers_list(self):
+        """
+        expected invalid_cfg
+        [u'ip nat inside source static 10.2.0.5 172.16.0.126 vrf'
+          ' nrouter-3ea5f9 redundancy neutron-hsrp-1064-3000',
+         u'ip nat inside source list neutron_acl_2564 pool'
+          ' nrouter-3ea5f9_nat_pool vrf nrouter-3ea5f9 overload',
+         u'ip nat pool nrouter-3ea5f9_nat_pool 172.16.0.124'
+          ' 172.16.0.124 netmask 255.255.0.0',
+         u'ip route vrf nrouter-3ea5f9 0.0.0.0 0.0.0.0'
+          ' Port-channel10.3000 172.16.0.1',
+         u'ip access-list standard neutron_acl_2564',
+         <IOSCfgLine # 83 'interface Port-channel10.2564'>,
+         <IOSCfgLine # 96 'interface Port-channel10.3000'>,
+         u'nrouter-3ea5f9']
+        """
+        cfg.CONF.set_override('enable_multi_region', False, 'multi_region')
+
+        router_db_info = []
+
+        self.config_syncer = aci_asr1k_cfg_syncer.ConfigSyncer(router_db_info,
+                                                      self.driver,
+                                                      self.hosting_device_info)
+        self.config_syncer.get_running_config = mock.Mock(
+            return_value=self._read_asr_running_cfg(
+                               'asr_basic_running_cfg_no_multi_region.json'))
+
+        invalid_cfg = self.config_syncer.delete_invalid_cfg()
+        self.assertEqual(8, len(invalid_cfg))
```

Comment:
can we assert against the real list instead of just the length of it. This way u can also remove the chunk of comments above as it might go out of date easily....Same thing for other UTs.